### DISCUSSION TITLE
Why doesn't directly modifying the gradient to re-render change the canvas?

### DISCUSSION THREAD

**Discussion by StringKe** [fabricjs/fabric.js#9529] [Q&A] [answered]
repo: https://codesandbox.io/p/devbox/adoring-gwen-forked-rryhp3

When the `colorStops` of a gradient object are updated, the new gradient will not be re-rendered even if `canvas.renderAll()` is actively called.

**Comment by ShaMan123** [comment, marked as answer]
It is expected, I agree.
But changing that will not mark the object as dirty. There are many cases as this. 
You need to `object.set({ dirty: true })`
